lmdeploy/pytorch/backends/moe.py

```diff
@@ -20,7 +20,7 @@ class SoftmaxTopKBuilder(ABC):
 
     @staticmethod
     @abstractmethod
-    def build(top_k: int, dim: int = -1):
+    def build(top_k: int, dim: int = -1, n_groups: int = -1):
```

Comment:
Dlinfer backend failed.

Reply:
Maybe it is better to add new method to SoftmaxTopKImpl, so we do not to enable the feature on init.
```
class SoftmaxTopKImpl:
    def __init__():
         self.n_groups=-1

    def set_n_groups(self, n_groups):
         self.n_groups = n_groups

```

So we don't have to update other backend